test/krew/krew.go

```diff
@@ -32,51 +33,82 @@ const krewBinaryEnv = "KREW_BINARY"
 
 // ITest is used to set up `krew` integration tests.
 type ITest struct {
-	t       *testing.T
-	plugin  string
-	args    []string
-	env     []string
-	tempDir *testutil.TempDir
+	t             *testing.T
+	plugin        string
+	pluginsBinDir string
+	args          []string
+	env           []string
+	tempDir       *testutil.TempDir
 }
 
 // NewTest creates a fluent krew ITest.
 func NewTest(t *testing.T) (*ITest, func()) {
 	tempDir, cleanup := testutil.NewTempDir(t)
-	pathEnv := setupPathEnv(t, tempDir)
+	binDir := setupKrewBin(t, tempDir)
+
 	return &ITest{
-		t:       t,
-		env:     []string{pathEnv, fmt.Sprintf("KREW_ROOT=%s", tempDir.Root())},
+		t:             t,
+		pluginsBinDir: binDir,
+		env: []string{
+			fmt.Sprintf("KREW_ROOT=%s", tempDir.Root()),
+			fmt.Sprintf("PATH=%s", augmentPATH(t, binDir)),
+		},
 		tempDir: tempDir,
 	}, cleanup
 }
 
-func setupPathEnv(t *testing.T, tempDir *testutil.TempDir) string {
-	krewBinPath := tempDir.Path("bin")
-	if err := os.MkdirAll(krewBinPath, os.ModePerm); err != nil {
+// setupKrewBin symlinks the $KREW_BINARY to $tempDir/bin and returns the path
+// to this directory.
+func setupKrewBin(t *testing.T, tempDir *testutil.TempDir) string {
+	krewBinary, found := os.LookupEnv(krewBinaryEnv)
+	if !found {
+		t.Fatalf("%s environment variable pointing to krew binary not set", krewBinaryEnv)
+	}
+	binPath := tempDir.Path("bin")
+	if err := os.MkdirAll(binPath, 0755); err != nil {
 		t.Fatal(err)
 	}
-
-	if krewBinary, found := os.LookupEnv(krewBinaryEnv); found {
-		if err := os.Symlink(krewBinary, tempDir.Path("bin/kubectl-krew")); err != nil {
-			t.Fatalf("Cannot link to krew: %s", err)
-		}
-	} else {
-		t.Logf("Environment variable %q was not found, using krew installation from host", krewBinaryEnv)
+	if err := os.Symlink(krewBinary, filepath.Join(binPath, "kubectl-krew")); err != nil {
+		t.Fatalf("cannot link krew binary: %s", err)
 	}
+	return binPath
+}
 
-	path, found := os.LookupEnv("PATH")
+// augmentPATH apprends the value to the current $PATH and returns the new
+// value.
+func augmentPATH(t *testing.T, v string) string {
+	curPath, found := os.LookupEnv("PATH")
 	if !found {
-		t.Fatalf("PATH variable is not set up")
+		t.Fatalf("$PATH variable is not set up, required to run tests")
 	}
 
-	return fmt.Sprintf("PATH=%s:%s", krewBinPath, path)
+	return v + string(os.PathListSeparator) + curPath
 }
 
-// Call configures the runner to call plugin with arguments args.
-func (it *ITest) Call(plugin string, args ...string) *ITest {
-	it.plugin = plugin
-	it.args = args
-	return it
+func (it *ITest) lookExecutable(file string) error {
```

Comment:
I know, you named that function like `exec.LookPath`. But it doesn't read well. How about

```suggestion
func (it *ITest) lookupExecutable(file string) error {
```